### PR TITLE
Notify teams about community PRs

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

Notifies teams by slack when a new community PR is open.

[details](https://datadoghq.atlassian.net/browse/ACIX-188).

### Additional Notes

Here is the message preview (`:pr:` will be replaced by the emote)

Screenshot 2024-04-26 at 10 34 18

### CODE REVIEW

File: tasks/github_tasks.py [hunk 1]
```diff
@@ -270,3 +279,34 @@ def assign_team_label(_, pr_id=-1):
...
+    # Get review teams / channels
+    gh = GithubAPI()
+
+    # Find teams corresponding to file changes
+    teams = _get_teams(gh.get_pr_files(pr_id)) or ['@datadog/agent-all']
+    channels = [GITHUB_SLACK_MAP[team.lower()] for team in teams if team if team.lower() in GITHUB_SLACK_MAP]
+
+    # Update labels
+    gh.add_pr_label(pr_id, 'community')
+    gh.add_pr_label(pr_id, 'team/triage')
```
Comment (amenasria) [inline tasks/github_tasks.py:297]: **💬 suggestion:** ‏Maybe this should be global variable in the file, I'm not a fan of hardcoding label names into function, wdyt ?

File: tasks/github_tasks.py [hunk 2]
```diff
@@ -270,3 +279,34 @@ def assign_team_label(_, pr_id=-1):
...
+
+    if teams != ['@datadog/agent-all']:
+        _assign_pr_team_labels(gh, pr_id, teams)
+
+    # Create message
+    gh = GithubAPI()
```
Comment (amenasria) [inline tasks/github_tasks.py:303]:
```suggestion

```

**🥜 nitpick:** ‏Duplicate of above

File: .github/workflows/external-contributor.yml [hunk 3]
```diff
@@ -11,12 +12,17 @@ jobs:
     runs-on: ubuntu-latest
     if: github.event.pull_request.head.repo.full_name != github.repository
     steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Install Python dependencies
```
Comment (chouetz) [inline .github/workflows/external-contributor.yml:17]: you can maybe use [setup-python](https://github.com/actions/setup-python#setup-python) here which can cache the dependencies (I think we must write it [like this](https://github.com/DataDog/datadog-agent/pull/24338/files#diff-76ad82686036d63b3bf52974dddfbe8fc72cd4172a0085d500190a3adcca3b41R30) to make it work)

File: tasks/github_tasks.py [hunk 4]
```diff
@@ -270,3 +279,34 @@ def assign_team_label(_, pr_id=-1):
...
+    pr = gh.get_pr(pr_id)
+    title = pr.title.strip()
+    pr_url = f'https://github.com/DataDog/datadog-agent/pull/{pr_id}'
+    message = f':pr: *New Community PR*\n{title} <{pr_url}|{repo}#{pr_id}>'
+
+    # Post message
+    client = slack.WebClient(os.environ['SLACK_API_TOKEN'])
```
Comment (amenasria) [inline tasks/github_tasks.py:310]: **❓ question:** ‏What would happen if the `SLACK_API_TOKEN` variable is not in the env?

File: tasks/github_tasks.py [hunk 5]
```diff
@@ -270,3 +279,34 @@ def assign_team_label(_, pr_id=-1):
...
+
+    if teams != ['@datadog/agent-all']:
+        _assign_pr_team_labels(gh, pr_id, teams)
+
+    # Create message
+    gh = GithubAPI()
+    pr = gh.get_pr(pr_id)
+    title = pr.title.strip()
+    pr_url = f'https://github.com/DataDog/datadog-agent/pull/{pr_id}'
+    message = f':pr: *New Community PR*\n{title} <{pr_url}|{repo}#{pr_id}>'
```
Comment (amenasria) [inline tasks/github_tasks.py:307]:
```suggestion
    message = f':pr: *New Community PR*\n{title}\n<{pr_url}|{repo}#{pr_id}>'
```

**💬 suggestion:** ‏Line might be super long without it, wdyt?

File: tasks/github_tasks.py (hunk 4 above)
Comment (amenasria) [reply tasks/github_tasks.py:310]: Not supposed to happen since it's defined in the job variables but we never know what could happen imo

**Review by chouetz — APPROVED** [review]

**Review by amenasria — APPROVED** [review]
**👏 praise:** ‏LGTM ! Thanks for this !

I left a few comments / suggestions.

File: tasks/github_tasks.py (hunk 1 above)
Comment (FlorentClarret) [reply tasks/github_tasks.py:297]: I agree, and same thing for `@datadog/agent-all` which is used several times

File: tasks/github_tasks.py [hunk 6]
```diff
@@ -270,3 +279,34 @@ def assign_team_label(_, pr_id=-1):
                 print(label_name, 'label assigned to the pull request')
             except github.GithubException.GithubException:
                 print(f'Failed to assign label {label_name}')
+
+
+@task
+def notify_community_pr(_, repo='', pr_id=-1):
```
Comment (FlorentClarret) [inline tasks/github_tasks.py:285]: **❓ question:** Why do we need the default values for the parameters?

**Review by FlorentClarret — COMMENTED** [review]
Left a few comments and agree with most of al the other comments. 

Thanks for this!

File: .github/workflows/external-contributor.yml [hunk 7]
```diff
@@ -1,3 +1,4 @@
+---
```
[the remark this answers is not in the document]
Comment (CelianR) [reply .github/workflows/external-contributor.yml:1]: It's a good practice to start a yaml document with this, usually it is not necessary but it's conventional

File: tasks/libs/requirements-github.txt [hunk 8]
```diff
@@ -8,3 +8,4 @@ cryptography==39.0.1
 PyJWT==2.4.0
 PyGithub==1.59.1
 toml~=0.10.2
+slackclient~=2.9.4
```
[the remark this answers is not in the document]
Comment (CelianR) [reply tasks/libs/requirements-github.txt]: I haven't seen that, thanks 👍

File: tasks/github_tasks.py (hunk 6 above)
Comment (CelianR) [reply tasks/github_tasks.py:287]: It is used to auto cast parameters to the target type with invoke (otherwise `pr_id` would be a `str`)

**Comment by CelianR** [comment]
I got a rebase problem, do not pay attention to this PR for the moment. Sorry for the notifications 😅

File: tasks/github_tasks.py [hunk 9]
```diff
@@ -260,6 +261,15 @@ def assign_team_label(_, pr_id=-1):
         print('No team found')
         return
 
+    _assign_pr_team_labels(gh, pr_id, teams)
+
+
+def _assign_pr_team_labels(gh, pr_id, teams):
+    """
+    Assign team labels (team/team-name) for each team (@dataog/team-name)
```
Comment (amenasria) [inline tasks/github_tasks.py:269]: **🥜 nitpick:** ‏
```suggestion
    Assign team labels (team/team-name) for each team (@datadog/team-name)
```

**Review by amenasria — APPROVED** [review]
LGTM ! 

**👏 praise:** ‏Nice one !

File: tasks/github_tasks.py [hunk 10]
```diff
@@ -274,3 +284,34 @@ def assign_team_label(_, pr_id=-1):
...
+    for label in os.environ['LABELS'].split(','):
+        gh.add_pr_label(pr_id, label)
+
+    if teams != [ALL_TEAMS]:
+        _assign_pr_team_labels(gh, pr_id, teams)
+
+    # Create message
+    pr = gh.get_pr(pr_id)
+    title = pr.title.strip()
+    pr_url = f'https://github.com/DataDog/datadog-agent/pull/{pr_id}'
```
[the remark this answers is not in the document]
Comment (pgimalac) [reply tasks/github_tasks.py:311]: I checked and the `PullRequest` object of `PyGithub` has a `url` member

[the remark this answers is not in the document]
Comment (kacper-murzyn) [reply tasks/github_tasks.py:311]: Note - for some reason in other places we use `html_url` field

[the remark this answers is not in the document]
Comment (pgimalac) [reply tasks/github_tasks.py]: Ah yes you are right, `url` is something like `https://api.github.com/repos/DataDog/datadog-agent/pulls/20000` while `html_url` is `https://github.com/DataDog/datadog-agent/pull/20000` so yeah `html_url` is the correct one here

**Pull request merged**